atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/SweepPriorityStoreImpl.java

```diff
@@ -101,7 +107,18 @@ private void tryInitialize() {
 
     private List<SweepPriority> loadPriorities(Transaction tx) {
         SweepPriorityTable table = sweepTableFactory.getSweepPriorityTable(tx);
-        return table.getAllRowsUnordered().transform(SweepPriorityStoreImpl::hydrate).immutableCopy();
+
+        // Load a single column first for each row. This is a much more efficient query on Cassandra
+        // than the full table scan that occurs otherwise.
+        List<SweepPriorityRowResult> rows = table.getRange(RangeRequest.builder()
+                .retainColumns(SweepPriorityTable.getColumnSelection(SweepPriorityNamedColumn.CELLS_DELETED))
+                .batchHint(READ_BATCH_SIZE)
+                .build())
+                .immutableCopy();
+
+        // Fetch all columns for the above rows directly
+        return Lists.transform(table.getRows(Lists.transform(rows, SweepPriorityRowResult::getRowName)),
```

Comment:
There is a small trade-off here in that we now make 2 requests to the underlying KVS, so this might not be faster for small tables, which we'd hope is usually the case (the product that spawned the ticket that needed this fix is an exception). It's probably fine though since a little bit slower on something that's already fast is usually OK

Reply:
Yep, and reads from this table shouldn't ever be blocking or performance critical.